### PR TITLE
[Feature]Use cn to execute broker and spark load

### PR DESCRIPTION
**Pull request by abc982627271**
## Problem Summary:
Fixes # (issue)

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 47 (48.94%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskScheduler.java|0|3|00.00%|[325, 326, 327]|
|:large_blue_circle:|com/starrocks/load/routineload/PulsarRoutineLoadJob.java|0|2|00.00%|[212, 213]|
|:large_blue_circle:|com/starrocks/common/util/PulsarUtil.java|0|3|00.00%|[141, 142, 149]|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|0|4|00.00%|[105, 106, 107, 110]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|2|7|28.57%|[544, 545, 547, 548, 549]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|3|8|37.50%|[511, 512, 513, 515, 699]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|1|2|50.00%|[233]|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|2|3|66.67%|[184]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|13|13|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/planner/FileScanNode.java [hunk 1]
```diff
@@ -144,7 +146,7 @@ public int compare(TBrokerFileStatus o1, TBrokerFileStatus o2) {
     private int filesAdded;
 
     // Only used for external table in select statement
-    private List<Backend> backends;
+    private List<ComputeNode> nodes;
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/planner/FileScanNode.java:149]: the above comment line needs an update?

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/planner/FileScanNode.java [hunk 2]
```diff
@@ -489,24 +491,37 @@ private void getFileStatusAndCalcInstance() throws UserException {
...
-            if (be.isAvailable()) {
-                backends.add(be);
+        nodes = Lists.newArrayList();
+
+        // TODO: need to refactor after be split into cn + dn
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
+            for (long cnId : warehouse.getAnyAvailableCluster().getComputeNodeIds()) {
+                ComputeNode cn = GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNode(cnId);
+                if (cn.isAvailable()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/planner/FileScanNode.java:509]:
```suggestion
                if (cn != null && cn.isAvailable()) {
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 20 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|10|18|55.56%|[506, 507, 508, 509, 510, 512, 513, 702]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|2|2|100.00%|[]|

**Review by kevincai — APPROVED** [review]

**Comment by gengjun-git** [comment]
@mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 20 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|10|18|55.56%|[505, 506, 507, 508, 509, 511, 512, 701]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 20 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|10|18|55.56%|[505, 506, 507, 508, 509, 511, 512, 701]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadJob.java|2|2|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**